sqlmesh/core/engine_adapter/base.py

```diff
@@ -456,11 +456,16 @@ def create_view(
             )
         )
 
-    def create_schema(self, schema_name: str, ignore_if_exists: bool = True) -> None:
+    def create_schema(
+        self, schema_name: str, catalog_name: t.Optional[str] = None, ignore_if_exists: bool = True
+    ) -> None:
         """Create a schema from a name or qualified table name."""
+        schema_name = schema_name.split(".")[0]
```

Comment:
Why do we continue to do the `split`? If we do, should we also anticipate 3-part names here?

Reply:
Nice catch!